hw/ip/edn/doc/programmers_guide.md

```diff
@@ -19,6 +19,47 @@ void edn_init(unsigned int enable) {
+Otherwise, EDN and the associated CSRNG application interface may get out of sync at which point both EDN and CSRNG have to be disabled and re-enabled.
+This means also unrelated CSRNG instances and EDNs get disabled and then need to be re-configured.
+
+The procedure for uninstantiating the associated CSRNG instance depends on the mode in which EDN is operating.
+- When running in **Software Port Mode**, firmware has to individually trigger commands via the [`SW_CMD_REQ`](registers.md#sw_cmd_req) register.
+  The status of the current command can be inferred using the [`SW_CMD_STS`](registers.md#sw_cmd_sts) register.
+  For details, refer to [Interaction with CSRNG Application Interface Ports](./theory_of_operation.md#interaction-with-CSRNG-application-interface-ports).
+  In case the current command is a `generate` command, firmware must wait for all entropy generated from this command to be consumed before the command is acknowledged by CSRNG (indicated by the `CMD_ACK` bit of the [`SW_CMD_STS`](registers.md#sw_cmd_sts) register being set).
+  Firmware can then issue an `uninstantiate` command to destroy the associated CSRNG instance.
+  Once the `uninstantiate` command is acknowledged, firmware can safely reconfigure EDN.
+
+- When running in **[Boot-Time Request Mode](./theory_of_operation.md#boot-time-request-mode)**, firmware must first de-assert the `BOOT_REQ_MODE` and `AUTO_REQ_MODE` fields in [`CTRL`](registers.md#ctrl) and then must wait for all boot-time entropy to be consumed.
+  EDN then automatically issues an `uninstantiate` command to destroy the associated CSRNG instance.
+  Upon completion, the state machine of EDN enters the `SWPortMode` state.
+  At this point, EDN has entered Software Port Mode (see above) and firmware can then safely reconfigure EDN.
+
+- When running in **[Auto Request Mode](./theory_of_operation.md#auto-request-mode)**, firmware must first de-assert the `AUTO_REQ_MODE` and `Boot_REQ_MODE` fields in [`CTRL`](registers.md#ctrl) and then must wait for the current command to complete, after which the state machine of EDN enters the `SWPortMode` state.
+  At this point, EDN has entered Software Port Mode (see above), and firmware can issue an `uninstantiate` command to destroy the associated CSRNG instance.
+  Once the `uninstantiate` command is acknowledged, firmware can safely reconfigure EDN.
+
+Notes:
+- Firmware can infer the EDN state using the [`MAIN_SM_STATE`](registers.md#main_sm_state) register.
+- The de-assertion of the `BOOT_REQ_MODE` and `AUTO_REQ_MODE` fields in [`CTRL`](registers.md#ctrl) is only registered by the EDN state machine upon completion of the current command.
+  In case of the `generate` command, this means to wait for all entropy generated from this command to have been consumed which depending on the configuration of the system, may take a very long time.
```

Comment:
Grammar nit: I think you want a comma after "which" (so that it pairs with the one after "system")